src/base/AsyncJobCalls.h

```diff
@@ -176,9 +176,18 @@ JobDialer<Job>::dial(AsyncCall &call)
     } catch (const std::exception &e) {
         debugs(call.debugSection, 3,
                call.name << " threw exception: " << e.what());
+        if (!job) {
+            debugs(call.debugSection, DBG_CRITICAL, "ERROR: Squid BUG: Job invalidated during " <<
+                   call.name << " that threw exception: " << e.what());
+            return; // see also: bug 4981, commit e3b6f15, and XXX in Http::Stream class description
+        }
         job->callException(e);
     }
 
+    if (!job) {
+        debugs(call.debugSection, DBG_CRITICAL, "ERROR: Squid BUG 4981: Job invalidated during " << call.name);
```

Comment:
Please do not put bug numbers into the logs. This tends towards an ongoing series of user reports when distros LTS continue to ship buggy Squid versions long after we fix it. Use a code comment instead.

Reply:
> Please do not put bug numbers into the logs. This tends towards an ongoing series of user reports when distros LTS continue to ship buggy Squid versions long after we fix it. Use a code comment instead.

I strongly disagree with that recommendation. IMO, bug reports about fixed bugs is a much smaller problem than a "BUG" message without attribution, both for admins and the Project. And that smaller problem can be reduced further by other means, such as improving Bugzilla rendering of closed bug reports.

Where applicable, we have been putting bug numbers into these BUG lines for a very long time and should continue to do so IMO. At the very least, until we develop consensus about this policy change, we should not ask authors to change (correct) bug attributions.